sdk/trace/batch_span_processor_test.go

```diff
@@ -257,3 +258,71 @@ func TestBatchSpanProcessorShutdown(t *testing.T) {
 	}
 	assert.Equal(t, 1, bp.shutdownCount)
 }
+
+func TestBatchSpanProcessorForceFlushSucceeds(t *testing.T) {
+	te := testBatchExporter{}
+	tp := basicTracerProvider(t)
+	option := testOption{
+		name: "default BatchSpanProcessorOptions",
+		o: []sdktrace.BatchSpanProcessorOption{
+			sdktrace.WithMaxQueueSize(0),
+			sdktrace.WithMaxExportBatchSize(3000),
+		},
+		wantNumSpans:   2053,
+		wantBatchCount: 1,
+		genNumSpans:    2053,
+	}
+	ssp := createAndRegisterBatchSP(option, &te)
+	if ssp == nil {
+		t.Fatalf("%s: Error creating new instance of BatchSpanProcessor\n", option.name)
+	}
+	tp.RegisterSpanProcessor(ssp)
+	tr := tp.Tracer("BatchSpanProcessorWithOption")
+	generateSpan(t, option.parallel, tr, option)
+
+	// Force flush any held span batches
+	err := ssp.ForceFlush(context.Background())
+
+	gotNumOfSpans := te.len()
+	spanDifference := option.wantNumSpans - gotNumOfSpans
+	if spanDifference > 10 || spanDifference < 0 {
+		t.Errorf("number of exported span not equal to or within 10 less than: got %+v, want %+v\n",
+			gotNumOfSpans, option.wantNumSpans)
+	}
```

Comment:
Why not test for equality here? Is there a reason we are looking to be within 10 of the expected?

Reply:
See [my comment above](https://github.com/open-telemetry/opentelemetry-go/pull/1757#issuecomment-811252211). Tests involving the queue operations here have been notoriously hard to make stable, particularly in CI environments with `GOMAXPROCS=1`.  My thinking here was that, unlike testing `Shutdown` where we really do need to ensure that all spans were exported, with `ForceFlush` we simply want to ensure that spans that had been queued were sent.  If one of the spans that was sent just before `ForceFlush` was called didn't get sent then it will get sent later when a timeout/full queue/subsequent flush or shutdown processes the queue again.  I don't think `ForceFlush` can be expected to be anything other than best-effort, so this seemed like a good compromise to avoid a flaky test while still ensuring it is doing the expected thing.